osconfig_tests/test_suites/package_management/package_management.go

```diff
@@ -112,44 +105,202 @@ func TestSuite(ctx context.Context, tswg *sync.WaitGroup, testSuites chan *junit
+		return
+	}
+
+	assign := &osconfigserver.Assignment{Assignment: assignment}
+
+	_, err = osconfigserver.CreateAssignment(ctx, logger, assign, parent)
+	if err != nil {
+		testCase.WriteFailure("error while creating assignment: \n%s\n", err)
+	}
+
+	defer cleanupAssignment(ctx, testCase, logger, assign)
+
+	client, err := daisyCompute.NewClient(ctx)
+	if err != nil {
+		testCase.WriteFailure("error creating client: %v", err)
+		return
+	}
+
+	testCase.Logf("Creating instance with image %q", testSetup.image)
+	testSetup.name = getTestSetupName(path.Base(testSetup.image), "packageinstalltest")
+	i := &api.Instance{
+		Name:        testSetup.name,
+		MachineType: fmt.Sprintf("projects/%s/zones/%s/machineTypes/n1-standard-1", testProject, testZone),
+		NetworkInterfaces: []*api.NetworkInterface{
+			&api.NetworkInterface{
+				Network: "global/networks/default",
+				AccessConfigs: []*api.AccessConfig{
+					&api.AccessConfig{
+						Type: "ONE_TO_ONE_NAT",
+					},
+				},
+			},
+		},
+		Metadata: &api.Metadata{
+			Items: []*api.MetadataItems{
+				testSetup.startup,
+			},
+		},
+		Disks: []*api.AttachedDisk{
+			&api.AttachedDisk{
+				AutoDelete: true,
+				Boot:       true,
+				InitializeParams: &api.AttachedDiskInitializeParams{
+					SourceImage: testSetup.image,
+				},
+			},
+		},
+		ServiceAccounts: []*api.ServiceAccount{
+			&api.ServiceAccount{
+				Email:  serviceAccountEmail,
+				Scopes: serviceAccountScopes,
+			},
+		},
+	}
+
+	inst, err := compute.CreateInstance(client, testProject, testZone, i)
+	if err != nil {
+		testCase.WriteFailure("Error creating instance: %v", err)
+		return
+	}
+	defer inst.Cleanup()
+
+	testCase.Logf("Waiting for agent install to complete")
+	if err := inst.WaitForSerialOutput("osconfig install done", 1, 5*time.Second, 5*time.Minute); err != nil {
+		testCase.WriteFailure("Error waiting for osconfig agent install: %v", err)
+		return
+	}
+
+	testCase.Logf("Agent installed successfully")
+
+	// agent might take a while to install the package. retry the command until timeout
+	var sleepTime = 5 * time.Second
+	var timeout = 60 * time.Second
+	var totalTime = 0 * time.Second
+	for {
+		sshCmd := getGcloudSshCmd(testZone, testSetup.name, dpkgListCmd, logger)
+		_, err := run(sshCmd, logger)
+
+		if err != nil {
+			testCase.WriteFailure("Error running verification command: %v", err)
+			return
+		}
+		// TODO refactor to remove hardcoding of package name
+		// read the serial console once
+		if err = inst.WaitForSerialOutput("cowsay", 1, sleepTime, sleepTime); err == nil {
+			return
+		}
+
+		totalTime = totalTime + sleepTime
+		if totalTime < timeout {
+			logger.Print("Did not get expected response from serial console, will retry")
+		} else {
+			testCase.WriteFailure("Error while assertion: %v", err)
+			break
+		}
+	}
+}
+
+func getGcloudSshCmd(zone string, instance string, pkgManagerCommand string, logger *log.Logger) *exec.Cmd {
```

Comment:
As discussed, don't shell out to gcloud, use startup scripts.
In the future if we need to ssh create an ssh client in go, don't rely on external commands.

Reply:
changed it to startup script